packages/medusa/src/models/batch-job.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  BeforeInsert,
+  Column,
+  CreateDateColumn,
+  DeleteDateColumn,
+  Entity,
+  PrimaryColumn,
+  UpdateDateColumn,
+} from "typeorm"
+import { ulid } from "ulid"
+import { BatchJobStatus, BatchJobType } from "../types/batch-job"
+import { DbAwareColumn, resolveDbType } from "../utils/db-aware-column"
+
+@Entity()
+export class BatchJob {
+  @PrimaryColumn()
+  id: string
+
+  @DbAwareColumn({ type: "enum", enum: BatchJobType })
+  type: BatchJobType
+
+  @DbAwareColumn({ type: "enum", enum: BatchJobStatus })
+  status: BatchJobStatus
+
+  @Column({ type: "varchar", nullable: true })
+  created_by: string | null
+
+  @DbAwareColumn({ type: "jsonb", nullable: true })
+  context: any
```

Comment:
issue: don't use any - potentially change to `Record<string, unknown>`?

Reply:
@srindom Woops, hadn't pushed my commit where I had changed it to Record<string, unknown> 😅  But on that note what are your thoughts on my note above regarding the typing of `context`?